### ISSUE TITLE
ModellingRule for displays located in an ObjectType will be ignored on atscm push

### ISSUE THREAD

**Issue by SedlatschekM** [atSCM/atscm#276]
*Insert error description here*
The ModellingRule of a display located in an ObjectType will be ignored on atscm push

**Steps to reproduce error**

 - Add ObjectType called "OT" in atvise
 - Add display to "OT"
 - atscm pull -> HasModellingRule will be created correct
 - Delete previous created display in atvise
 - atscm push

**Original error message**

Display will be created successfully, but the ModellingRule is now set to "None" in atvise - previous the ModellingRule was set to "Shared".
 
---

<details><summary><strong>Environment</strong></summary>

| package       |               version |
| :------------ | --------------------: |
| OS            | *Windows* |
| atscm         |         0.7.0 | 
| atscm-cli     |     1.0.0-beta.24 | 
| atvise server | 3.2.2 |
| node          |          10.15.3 | 
| npm           |           6.4.1 | 
 
</details>

<details><summary><strong>Project configuration</strong></summary>

nothing

</details>

**Comment by breussm** [comment]
@SedlatschekM  please verify this with current version 1.0.0.26Beta

**Comment by SedlatschekM** [comment]
@breussm Test failed with current version 1.0.0-beta.27 - atscm pull throw error on node "ObjectTypes.PROJECT.OT" cause of undefined "value" property

Error:
![image](https://user-images.githubusercontent.com/31237004/54683786-da91e600-4b12-11e9-8a01-3043509d71c1.png)

Configuration:
![image](https://user-images.githubusercontent.com/31237004/54683841-feedc280-4b12-11e9-9aa1-d50f4f7b9dc4.png)

**Comment by LukasHechenberger** [comment]
Hm, cannot reproduce that right now...

I added an integration test doing just what you described:

 - Pull an ObjectType *OT* with a child display (*TestDisplay* in my case), that has the modelling rule *Shared*
 - Delete it on the server
 - Push it again

It works in my local dev environment (macOS, node 8 and 10) as well as on [CircleCI](https://circleci.com/workflow-run/fb725012-be83-499f-934c-cb74b18f43de) (Linux, node 8) and [Appveyor](https://ci.appveyor.com/project/LukasHechenberger/atscm/builds/23244073https://ci.appveyor.com/project/LukasHechenberger/atscm/builds/23244073) (Windows, node 8).

Could you try running the test against your setup? Steps required:

-  Clone this repo (Or update if already cloned: `git fetch --all`)
-  Checkout the `fix-276-value-undefined` branch `git checkout fix-276-value-undefined`
-  Edit host and port inside `./test/fixtures/Atviseproject.babel.js` to match your atserver
-  Run `npm run -s test:base -- test/integration/issues/issue-276.spec.js`

**Comment by SedlatschekM** [comment]
Hi,

I followed your described steps and got the following result:

![image](https://user-images.githubusercontent.com/31237004/55554863-1e284a80-56e4-11e9-9fc7-6e2e476a5015.png)

**Comment by LukasHechenberger** [comment]
Okay, weird... Seems like you installed a wrong version of *node-opcua*. Sadly they don't follow semantic versioning, therefore it's required you install the locked version specified inside `./package-lock.json` (currently *0.0.60*).
This should be done by default with npm v6. You sure you didn't use an older version when you ran `npm install`? You can check the actual version inside `./node_modules/node-opcua/package.json`.

To ensure the right node modules you can run:
```bash
# Reset changes in package lock (if any)
git checkout -- package-lock.json
# Clean install node modules (removes all modules first, doesn't modify lockfile)
npm ci
```

**Comment by SedlatschekM** [comment]
Thank you for this information - I execute the steps below once again.

    Clone this repo (Or update if already cloned: git fetch --all)
    Checkout the fix-276-value-undefined branch git checkout fix-276-value-undefined
    Edit host and port inside ./test/fixtures/Atviseproject.babel.js to match your atserver
    Run npm run -s test:base -- test/integration/issues/issue-276.spec.js

And got the following result:

![image](https://user-images.githubusercontent.com/31237004/55866825-71394c00-5b81-11e9-98bd-4f99ed35226b.png)

**Comment by LukasHechenberger** [comment]
Did it again on my mac, as well as on my win vm, still working as expected...

What i did (in an empty folder):
```
git clone https://github.com/atSCM/atscm
cd atscm
git checkout fix-276-value-undefined
npm install
npm run -s test:base -- test/integration/issues/issue-276.spec.js
```
I forgot to mention the `npm install`-step earlier, but I guess if you missed this one the tests woundn't even have started.

As far as I can tell, it works on macOS and Windows 10, with node v8, 9 and 10 and npm v5 and 6...

**Console output (Win 10, node 9, npm 5):**
```
C:\Users\lukas\Downloads\tmp\issue-276>git clone https://github.com/atSCM/atscm
Cloning into 'atscm'...
remote: Enumerating objects: 503, done.
remote: Counting objects: 100% (503/503), done.
remote: Compressing objects: 100% (280/280), done.
remote: Total 20320 (delta 300), reused 364 (delta 198), pack-reused 19817
Receiving objects: 100% (20320/20320), 34.85 MiB | 2.28 MiB/s, done.
Resolving deltas: 100% (14352/14352), done.

C:\Users\lukas\Downloads\tmp\issue-276>git checkout fix-276-value-undefined
fatal: Not a git repository (or any of the parent directories): .git

C:\Users\lukas\Downloads\tmp\issue-276>cd atscm

C:\Users\lukas\Downloads\tmp\issue-276\atscm>git checkout fix-276-value-undefined
Switched to a new branch 'fix-276-value-undefined'
Branch 'fix-276-value-undefined' set up to track remote branch 'fix-276-value-undefined' from 'origin'.

C:\Users\lukas\Downloads\tmp\issue-276\atscm>npm install

> uws@0.14.5 install C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\uws
> node-gyp rebuild > build_log.txt 2>&1 || exit 0


> husky@0.14.3 install C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\husky
> node ./bin/install.js

husky
setting up Git hooks
done


> node-opcua@0.0.60 postinstall C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua
> node postinstall.js

Executing  node bin/generate_opcua_classes.js --clear
done
done ... (0)
Executing  node bin/crypto_create_CA.js demo --silent

        Create Demo certificates


reading configuration
=====================

 Creating default config file  C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua\certificates\config.js
 configuration =
   commonName                     : NodeOPCUA-COMMONNAME
   organization                   : NodeOPCUA
   organizationUnit               : Unit
   locality                       : Paris
   state                          : IDF
   country                        : FR
   validity                       : 5475
   keySize                        : 2028
   certificateDir                 : C:/Users/lukas/Downloads/tmp/issue-276/atscm/node_modules/node-opcua/certificates
   CAFolder                       : C:/Users/lukas/Downloads/tmp/issue-276/atscm/node_modules/node-opcua/certificates/CA
   PKIFolder                      : C:/Users/lukas/Downloads/tmp/issue-276/atscm/node_modules/node-opcua/certificates/PKI
creating openssl_folder C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl
checking presence of  C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\openssl.exe
openssl seems to be missing and need to be installed
downloading http://indy.fulgan.com/SSL/openssl-1.0.2j-x64_86-win64.zip
deflating  C:\Users\lukas\AppData\Local\Temp\openssl-1.0.2j-x64_86-win64.zip
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\HashInfo.txt
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\libeay32.dll
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\OpenSSL License.txt
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\openssl.exe
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\ReadMe.txt
 unzipping : C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\ssleay32.dll
unzip done
verifying  true OK  C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\openssl.exe
done
checking presence of  C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\openssl.exe
        stdout OpenSSL 1.0.2j  26 Sep 2016
openssl_path C:\Users\lukas\Downloads\tmp\issue-276\atscm\node_modules\node-opcua-pki\bin\openssl\openssl.exe
     hostname =  LUKASHECHENFA2E
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-Client
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-Client
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-Server
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-Server
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-DiscoveryServer
 urn =  urn:LUKASHECHENFA2E:NodeOPCUA-DiscoveryServer
done ... (0)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.1 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\@babel\cli\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1865 packages in 939.232s


   ╭─────────────────────────────────────╮
   │                                     │
   │   Update available 5.6.0 → 6.9.0    │
   │       Run npm i npm to update       │
   │                                     │
   ╰─────────────────────────────────────╯


C:\Users\lukas\Downloads\tmp\issue-276\atscm>npm run -s test:base -- test/integration/issues/issue-276.spec.js


  Issue #276 (https://github.com/atSCM/atscm/issues/276)
    Modelling rule should be mapped correctly
      √ should recreate all fields (484ms)


  1 passing (3s)


C:\Users\lukas\Downloads\tmp\issue-276\atscm>node -v
v9.4.0

C:\Users\lukas\Downloads\tmp\issue-276\atscm>npm -v
5.6.0
```

**Comment by breussm** [comment]
@SedlatschekM can you try this again?
